### PR TITLE
docs: Add doc notes regards tf v1.3 support

### PR DESCRIPTION
**Pull request by Zvikan**
### What does this PR do?

🛑 Please open an issue first to discuss any significant work and flesh out details/direction - we would hate for your time to be wasted.
Consult the [CONTRIBUTING](https://github.com/aws-ia/terraform-aws-eks-blueprints/blob/main/CONTRIBUTING.md#contributing-via-pull-requests) guide for submitting pull-requests.

- Adding notes regards terraform version 1.3 support
### Motivation

- Users starting to use TF version 1.3 which we don't support yet, as a result we should start with adding a clear message about it.

### More

- [ ] Yes, I have tested the PR using my local account setup (Provide any test evidence report under Additional Notes)
- [ ] Yes, I have added a new example under [examples](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/examples) to support my PR
- [ ] Yes, I have created another PR for add-ons under [add-ons](https://github.com/aws-samples/eks-blueprints-add-ons) repo (if applicable)
- [ ] Yes, I have updated the [docs](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/docs) for this feature
- [ ] Yes, I ran `pre-commit run -a` with this PR

**Note**: Not all the PRs require a new example and/or doc page. In general:
- Use an existing example when possible to demonstrate a new addons usage
- A new docs page under `docs/add-ons/*` is required for new a new addon

### For Moderators

- [ ] E2E Test successfully complete before merge?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -105,6 +105,9 @@ If you are interested in contributing to EKS Blueprints, see the [Contribution g
 
 ---
 
+> **_NOTE:_** Use terraform versions after version 1.0.0 and before version 1.3.0.
```
Comment (bryantbiggs) [inline README.md:108]: terraform-docs will wipe this out (after CI checks complain)

Comment (Zvikan) [reply README.md:108]: it's before the `<!-- BEGINNING OF PRE-COMMIT-TERRAFORM DOCS HOOK -->` , you sure?

Comment (bryantbiggs) [reply README.md:108]: oh .... I'm not that smart, please  ignore me 😬

**Comment by askulkarni2** [comment]
Likely not needed after #1011

**Comment by csantanapr** [comment]
Using terraform 1.3 is still a problem we need to document for user, they can't use 1.3 cc @bryantbiggs

**Comment by csantanapr** [comment]
don't know why the md lint is failing is just one line change

**Comment by bryantbiggs** [comment]
its this https://github.com/aws-ia/terraform-aws-eks-blueprints/pull/1033/commits/7abef5be24e9718c1b9f350b5ca08595a645cfd9

we can ignore for now and once my PR lands it will be fixed

**Comment by csantanapr** [comment]
ah like something got moved, and the current md is broken in main
@bryantbiggs how do we ignore it? to be able to merge it?

**Review by bryantbiggs — APPROVED** [review]

**Pull request merged**